### DISCUSSION TITLE
Image processing PHP

### DISCUSSION THREAD

**Discussion by val-rot** [pvtl/docker-dev#26] [General]
HI,

I wanted to install imagemagick or graphicsmagick, but unfortunately this didn't work.

![grafik](https://user-images.githubusercontent.com/24671073/222181789-c98565fe-94bb-40f1-b730-dfff414441f7.png)

these are often used in web development and many cms systems. It would be great if that was included beforehand. Any tips on how I could install it?

**Comment by wireblue** [comment]
Hi @val-rot,

Your config looks ok at a glance. What errors are you seeing?

If you docker-exec into the PHP container, can you run the commands manually to install and configure imagemagick?

Once you're figured out the commands, you can add them to the `EXTRA_CMDS_PHP81` environment variable so they're run each time you update your docker-dev containers.

**EDIT**
If you're getting a `The following signatures were invalid` error for Yarn while trying to install your packages, then use this solution:
https://github.com/yarnpkg/yarn/issues/7866

I'll look at including this fix into our published docker images soon.

**Reply by val-rot** [reply]
Hello @wireblue ,

Thank you for the info.

I have now installed GM & IM manually.

sudo apt-get update
sudo apt-get install graphicsmagick
sudo apt-get install imagemagick

//--- php.ini
extension=imagick.so
extension=gmagick.so


Of course I would like to automate this. do you have an example for me how to do that with EXTRA_CMDS_PHP81?

Incidentally, the post from above did not issue any messages, neither errors nor anything else. I can't explain that either.

But it is also relatively quick via console.

**Reply by wireblue** [reply]
Ahh, great to hear.

In your .env file you can set something like:

```
EXTRA_CMDS_PHP81=sudo apt-get update && sudo apt-get install imagemagick graphicsmagick ...
```

Just adjust the command to suit your circumstances.

You can add any php.ini changes like so:
https://github.com/pvtl/docker-dev/blob/master/docs/faqs.md#adding-custom-php-configuration

**Reply by val-rot** [reply]
Here my result.

![grafik](https://user-images.githubusercontent.com/24671073/223458413-61603d58-e9fe-4b3a-80b7-f5623ba00149.png)

This error come always, i have change kombination and set only this step by step (sudo apt-get update) only stay single. But dont works with this on my side.
